README.md

```diff
@@ -221,18 +221,20 @@ ReSub’s implementation of this method always returns true, which is inline wit
 
 Subclasses should implement some or all of the following methods:
 
-##### `protected _buildState(props: P, initialBuild: boolean): S`
+##### `protected _buildState(props: P, initialBuild: boolean, incomingState: {} | S): Partial<S> | undefined`
 
 This method is called to rebuild the module’s state. All but the simplest of components should implement this method. It is called on three occurrences:
 
 1. During initial component construction, `initialBuild` will be true. This is where you should set all initial state for your component. This case rarely needs special treatment because the component always rebuilds all of its state from its props, whether it's an initial build or a new props received event.
-2. In the React lifecycle, during a `getDerivedStateFromProps`, this is called on *every* render.
+2. In the React lifecycle, during a `getDerivedStateFromProps`, this is called before *every* render.
 3. When this component subscribes to any stores, this will be called whenever the subscription is triggered. This is the most common usage of subscriptions, and the usage created by autosubscriptions.
 
 Any calls from this method to store methods decorated with `@autoSubscribe` will establish an autosubscription.
 
 React’s `setState` method should not be called directly in this function. Instead, the new state should be returned and `ComponentBase` will handle the update of the state.
 
+Note: If your _buildState ever relies on component state, utilize the incomingState argument, otherwise you risk using an old snapshot of the state (See https://github.com/microsoft/ReSub/issues/131 for more details)
```

Comment:
I feel like this might be noted here - https://github.com/microsoft/ReSub/blob/master/README.md#L9 - when I do breaking changes I usually have a separate migration page linked in changelog + readme for each major version change where I give an example of what to look for, and how to change it. Otherwise I get bombed with issues later - pay me now / pay me later. $0.02

Reply:
This is separate but possibly also worth noting I saw my react migration tidbits in there but I'm not sure they are valid anymore? 'UNSAFE_' anything - https://github.com/microsoft/ReSub/blob/master/README.md#L370